sys/kern/sys_process.c

```diff
@@ -1534,43 +1541,56 @@ kern_ptrace(struct thread *td, int req, pid_t pid, void * __capability addr, int
 #endif
 
 	case PT_LWPINFO:
-		if (data <= 0 ||
-#ifdef COMPAT_FREEBSD32
-		    (!wrap32 && data > sizeof(*pl)) ||
-		    (wrap32 && data > sizeof(*pl32))) {
-#else
-		    data > sizeof(*pl)) {
-#endif
+	{
+		bool set_siginfo;
+
+		if (data <= 0) {
 			error = EINVAL;
 			break;
 		}
+#ifdef COMPAT_FREEBSD64
+		if (wrap64) {
```

Comment:
I eventually want to push this out into freebsd64_ptrace(), but that seemed too ugly to do now.  I will eventually clean up the freebsd32/64 bits to move them out of kern_ptrace() and into the callers as much as possible, but that will be a future cleanup.